main/ahb_for_sles.go

```diff
@@ -97,13 +98,49 @@ func parseCfg(filename string) (map[string]map[string]string, error) {
 	return ini, nil
 }
 
-func _isRegistered() (bool, error) {
-	services, err := filepath.Glob("/etc/zypp/services.d/*.service")
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
-		return false, err
+func _getSUSEConnectStatus() (bool, bool, error) {
+	commandOutput, error := RunShellCommand(0, "SUSEConnect", "-s")
+	if error != nil {
+		fmt.Println(error)
+		return false, false, error
+	}
+	var suseConnectStatus []map[string]interface{}
+	json.Unmarshal([]byte(commandOutput), &suseConnectStatus)
+	subscriptionStatus := fmt.Sprintf("%v", suseConnectStatus[0]["subscription_status"])
+	status := fmt.Sprintf("%v", suseConnectStatus[0]["status"])
+	if strings.ToLower(status) == "registered" && strings.ToLower(subscriptionStatus) == "active" {
+		// system is registered with an active subscription
+		// check if services are present
+		services, err := filepath.Glob("/etc/zypp/services.d/*.service")
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return true, true, err
+		}
+		if len(services) == 0 {
+			errorMessage := "There are no services. Please, re register the system"
```

Comment:
The system is already registered, if we tell the user to re-register than 2 subscriptions will be counted against one system. That is bad advice at best. We need to tell the user the correct action to take, or restore the services. The service restore is fairly straight forward:

```
SUSEConnect --list-extensions
```
Then loop through every product that has already been `(Activated)` or said differently for every line in the output that has "Deactivate with:" take the presented `SUSEConnect` command and drop the `-d`

Reply:
Done, also activate Public Cloud module in case it was not active before